fastatophy/fastatophy.go

```diff
@@ -0,0 +1,112 @@
+// Copyright ©2017 The bíogo Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// fastatophy converts a multiple-sequence alignment in
+// FASTA to PHYLIP (sequential) format.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"github.com/biogo/biogo/alphabet"
+	"github.com/biogo/biogo/io/seqio"
+	"github.com/biogo/biogo/io/seqio/fasta"
+	"github.com/biogo/biogo/seq/linear"
+)
+
+var (
+	inf  = flag.String("in", "", "input FASTA filename")
+	outf = flag.String("out", "", "output PHYLIP filename")
+	help = flag.Bool("help", false, "help prints this message")
+)
+
+func main() {
+	flag.Parse()
+	if *help {
+		flag.Usage()
+		os.Exit(0)
+	}
+
+	var in *os.File
+	var r *fasta.Reader
+	var err error
+	t := linear.NewSeq("", nil, alphabet.Protein)
+	if *inf == "" {
```

Comment:
This should happen before here, just after the help check, and exit(1) if `*inf == ""`.

```
var (
	inf  = flag.String("in", "", "input FASTA filename (required)")
	outf = flag.String("out", "", "output PHYLIP filename")
	help = flag.Bool("help", false, "help prints this message")
)
func main() {
	flag.Parse()
	if *help {
		flag.Usage()
		os.Exit(0)
	}
	if *inf == "" {
		flag.Usage()
		os.Exit(1)
	}

	in, err := os.Open(*inf)
	if err != nil {
		log.Fatalf("failed to open FASTA file %q: %v", *inf, err)
	}
	defer in.Close()
	r := fasta.NewReader(in, 	linear.NewSeq("", nil, alphabet.Protein))
	...
```